system-contract-dapp-playground/src/sections/landing/index.tsx

```diff
@@ -21,10 +21,79 @@
  */
 
 import { motion } from 'framer-motion';
+import { useToast } from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+import { useRouter } from 'next/navigation';
+import { storeAccountsInCookies } from '@/api/cookies';
 import { VerticalCommonVariants } from '@/libs/framer-motion/variants';
+import { requestAccount, isCorrectHederaNetwork, getWalletProvider } from '@/api/wallet';
+import { NoEthToast, CommonErrorToast, NetworkMismatchToast } from '@/components/toast/CommonToast';
 
 const LandingPage = () => {
+  const router = useRouter();
+  const toaster = useToast();
+  const [accounts, setAccounts] = useState<string[]>([]);
   const verticalVariant = VerticalCommonVariants(30, 0.5);
+  const { walletProvider, err: walletProviderErr } = getWalletProvider();
+
+  /** @dev handle connect wallet when a user click `connect wallet` button */
+  const handleConnectWallet = async () => {
+    // handle ethereum object or walletProvider being null by toasting it out on the client
+    if (walletProviderErr === '!ETHEREUM' || !walletProvider) {
+      NoEthToast({ toaster });
+      return;
+    }
+
+    // detect if the current network is expected Hedera Networks
+    if (!(await isCorrectHederaNetwork(walletProvider))) {
+      NetworkMismatchToast({ toaster });
+      return;
+    }
+
+    // call requestAccount() API for users to connect their account to the DApp
+    const { accounts, err: getAccountErr } = await requestAccount(walletProvider!);
+
+    // handle getAccountError
+    if (getAccountErr || !accounts || accounts.length === 0) {
+      let errorMessage = 'Unknown error appeared...';
+
+      if (JSON.stringify(getAccountErr).indexOf('4001') !== -1) {
```

Comment:
Q: why index 4001? Are you searching for the error code?
If so add a comment

Reply:
Oh yeah these are special indexes from Metamask. https://docs.metamask.io/wallet/reference/provider-api/#errors 
I'll add a comment to these! I really thought I did tho...